### PR TITLE
Rename Scanner module to TypeAnnotatorScanner.

### LINKED ISSUE #25: Rename Scanner module to NullAwayAnnotatorScanner
**Describe the task**
As @lazaroclapp suggested, the name `Scanner` is too generic, specifically when mentioned along other `ErrorProneCheckers` and should be refined. Currently the suggestion is: `NullAwayAnnotatorScanner`.

**Expected behavior**
Rename of module `Scanner` to `NullAwayAnnotatorScanner` with follow ups of all required updates.

**Screenshots**
Not-Applicable.

**Additional context**
Not-Applicable.

### PR DESCRIPTION
**Pull request by nimakarimipour**
This PR resolves #25 and renames `Scanner` module to `AnnotatorScanner` module.

The name `Scanner` is too generic, specifically when mentioned along other `ErrorProneCheckers` and should be refined. Currently the suggestion is: `AnnotatorScanner`.

This involved changes in the configurations, unit tests project templates, and install scripts.

@lazaroclapp Previously you suggested `NullAwayAnnotatorScanner`, however, this checker is not related to NullAway and I am intending to apply the same technique and use this whole repo in general to infer other type systems. Given that this repo is called `NullAwayAnnotator`, I still think it makes sense to generalize this name. Please let me know your thoughts on this and if you think I should rename it to `NullAwayAnnotatorScanner`. Thanks :)

----
**Update**:
The final decision regarding name is: `TypeAnnotatorScanner`.

### CODE REVIEW

**Comment by lazaroclapp** [comment]
Haven't looked at the specific rename changes, but regarding the `AnnotatorScanner` name itself...

> Previously you suggested `NullAwayAnnotatorScanner`, however, this checker is not related to NullAway and I am intending to apply the same technique and use this whole repo in general to infer other type systems. Given that this repo is called `NullAwayAnnotator`, I still think it makes sense to generalize this name.

My problem here is not really within the context of this repo. `Scanner` would be fine if the checker name only ever appeared in the context of this repository or as dependency artifact name (i.e. `edu.ucr.cs.riple.nullawayannotator:scanner:$someVersion`). The problem is that the simple name is also used as the name of the checker in Error Prone: `-Xep:AnnotatorScanner:[...]` becomes part of our configuration, which is file configuring a few dozen checkers - most of them part of Error Prone itself. So my concern is being able to distinguish it from some internal EP checker within that list or (more generally and less Uber specific!) within the arguments passed to `javac`. Clearly this isn't part of some general Error Prone Annotator, at least not for the time being.

I would consider `UCRAnnotatorScanner`, if you don't want to tie it too closely to NullAway. That said, my hope is still that we can get to the point of merging all pieces of the auto-annotator into the uber/NullAway repo at some point and release them together, so, not sure you really can or want to distance yourself from the NullAway brand here. Thoughts, @msridhar ?

Another thing we could do is ask Error Prone folks (i.e. raise an issue on their tracker) about whether there is any sort of name-spacing convention that is used or can be used for third-party Error Prone checkers. But I, personally, am not aware of one.

File: runner/README.md
Comment (ketkarameya) [inline runner/README.md:21]:
```suggestion
"-Xep:AnnotatorScanner:ERROR", // to activate AnnotatorScanner
```

File: README.md
Comment (ketkarameya) [inline README.md:123]:
```suggestion
It installs `AnnotatorScanner` checker in `maven local` and creates the jar file of `core` module and moves it to the correct path. To run it, execute the command below:
```

**Review by ketkarameya — APPROVED** [review]
LGTM

**Comment by msridhar** [comment]
Hey, so bottom line, I think we need a better name than `AnnotatorScanner`.  What about `CodeIndexSerializer`?  I think what this checker does is serialize some indexing information about the code, like where fields are used, where methods are called, etc.  Or maybe `TypeAnnotatorIndexer`?  @lazaroclapp we do have concrete plans to apply this technique to more than just NullAway, so I think keeping the naming separate and possibly the repositories may be a good idea for now, at least until we can have a more detailed discussion about future plans.

**Review by msridhar — DISMISSED** [review]
Would like the naming discussion resolved before this lands, if possible

File: runner/README.md
Comment (nimakarimipour) [reply runner/README.md]: 54540be0a34cd643e6b8cf733250e9a8de7fbfd7

File: README.md
Comment (nimakarimipour) [reply README.md]: 54540be0a34cd643e6b8cf733250e9a8de7fbfd7

**Review by msridhar — COMMENTED** [review]
No time to review and approve all these changes but I don't object

File: core/src/test/java/edu/ucr/cs/riple/core/ConfigurationTest.java
Comment (lazaroclapp) [inline core/src/test/java/edu/ucr/cs/riple/core/ConfigurationTest.java:121]: Was this for debugging? I don't think the tests should print to stdout on every run, and we are already checking the message below on a successful run. Let's remove this line.

File: library-model-loader/src/main/resources/edu/ucr/cs/riple/librarymodel/nullable-methods.tsv
Comment (lazaroclapp) [inline library-model-loader/src/main/resources/edu/ucr/cs/riple/librarymodel/nullable-methods.tsv:2]: Why are re removing this library model TSV? (Just want to understand, since it seems unrelated to the renaming and it isn't called out in the PR description)

File: runner/start.sh
Comment (lazaroclapp) [inline runner/start.sh:26]: Why isn't this ending in a newline?

File: type-annotator-scanner/README.md
Comment (lazaroclapp) [inline type-annotator-scanner/README.md:1]: At some point we might want proper documentation here. Even now, I wouldn't mind a single-paragraph description of what the Type Annotator Scanner does and what it is used for here 🙂 (don't spend a lot of time on it, but it's certainly worth 5-10 minutes to explain its overall purpose)

File: core/src/test/java/edu/ucr/cs/riple/core/ConfigurationTest.java
Comment (nimakarimipour) [reply core/src/test/java/edu/ucr/cs/riple/core/ConfigurationTest.java]: Yes exactly, this is left out code from debugging. bd21c3d0e4abd6d8e120b7bc20523056e5187e0b

File: library-model-loader/src/main/resources/edu/ucr/cs/riple/librarymodel/nullable-methods.tsv
Comment (nimakarimipour) [reply library-model-loader/src/main/resources/edu/ucr/cs/riple/librarymodel/nullable-methods.tsv:2]: `LibraryModelsLoader` module here is used for unit tests and these are left over contents from previous runs. They shouldn't appear in the git repository.

File: runner/start.sh
Comment (nimakarimipour) [reply runner/start.sh]: 8f79e985a4d0f416d63cd465a247ac3ffe11a573

File: type-annotator-scanner/README.md
Comment (nimakarimipour) [reply type-annotator-scanner/README.md]: Sure, I added an overview of the module. Will complete it later :)
048a3bb4e86ee5417677840774efb91dda4f5c70

File: type-annotator-scanner/README.md
Comment (lazaroclapp) [inline type-annotator-scanner/README.md:41]: "This data" (uncountable noun)

**Review by lazaroclapp — APPROVED** [review]
LGTM! (After the change to the TSV file lands, can we make a follow up PR to `.gitignore` that file? Or otherwise make sure that the tests clean up their changes to that file after running, whether or not they succeed?)

**Comment by nimakarimipour** [comment]
> Other than a minor doc nit below, this LGTM! (After the change to the TSV file lands, can we make a follow up PR to `.gitignore` that file? Or otherwise make sure that the tests clean up their changes to that file after running, whether or not they succeed?)

Thank you for the review. Sure will resolve the `tsv` file in a follow up PR.

File: type-annotator-scanner/README.md
Comment (nimakarimipour) [reply type-annotator-scanner/README.md]: 1e2286bae6c6cc4f36c86d81d760df0f80c294fd

**Comment by nimakarimipour** [comment]
@ketkarameya I can see you already approved this PR. Will land it as soon as the checks pass. Thank you.

**Pull request merged**
